www/_blog/2021-07-27-storage-beta.mdx

```diff
@@ -0,0 +1,167 @@
+
+## Context menus
+
+The storage explorer in our Dashboard feels like using a file explorer in your operating system. There were a lot of challenges to make it feel intuitive and robust, since S3 is a not a traditional filesystem. For example, S3 doesn't support a move operation natively - you need to copy the object to the new location and delete the original object. S3 also doesn't have a concept of directory, which means there is no API call to rename a directory. But to make the storage explorer feel like a native file explorer, we implemented these features.
+
+![Context menu 1](/new/images/blog/launch-week-sql-day-2/context-menu-1.png)
+
+![Context menu 2](/new/images/blog/launch-week-sql-day-2/context-menu-2.png)
+
+### The case of the disappearing folders
+
+As we already mentioned, S3 doesn't really have a concept of directories. So if a user uploads an object with path `dir/subdir/avatar.png`, the dir and subdir are automatically "created" and there is no explicit API call to create them.
+
+This leads to an issue where if a user creates an empty directory and refreshes the page, the directory disappears since it is not persisted to S3. To simulate a normal filesystem (where empty directories are not garbage collected), we upload an empty file when the user creates a directory on the dashboard. The dashboard hides this placeholder file and the placeholder is deleted when the user uploads an object in the directory - voila, no more disappearing directores!
+
+## Performance improvements for large buckets
+
+We significantly improved rendering for a large number of items in the Dashboard. We use a virtualization library to render the objects in view and actively discard DOM elements which are not currently rendered from memory. This makes scrolling through thousands of items buttery smooth.
+
+Here is a demo testing rendering the heaviest object known to mankind - `node_modules`!
+
+<video width="99%" autoPlay="autoplay" muted playsInline controls={true}>
+  <source
+    src="/new/videos/blog/launch-week-storage-2/large-buckets.mp4"
+    type="video/mp4"
+    muted
+    playsInline
+  />
+</video>
+
+## API changes
+
+Supabase Storage now supports `upsert`. Shoutout to [@ankitjena](https://github.com/ankitjena) for
+[this Pull Request](https://github.com/supabase/storage-api/pull/32).
+
+![Storage now supports upserts](/new/images/blog/2021-june/supabase-storage-upsert.png)
+
+We also have automated API docs generated [here](https://supabase.github.io/storage-api/#/) if you want to use the Storage API directly.
+
+### Storage in the wild
+
+We have a few production applications using Supabase Storage. Wataru is using Storage to power his community site [Shikutoku](https://shikutoku.me/) and Tyler is using Supabase Storage to store short video clips uploaded to his Spot app built with Flutter
```

Comment:
Add link to Spot case study:

https://supabase.io/blog/2021/07/27/spot-flutter-with-postgres